kratos/tests/cpp_tests/processes/test_compute_nodal_gradient_process.cpp

```diff
@@ -258,22 +268,183 @@ namespace Kratos
-            typedef ComputeNodalGradientProcess<3, Variable<double>, Historical> GradientType;
+            typedef ComputeNodalGradientProcess<ComputeNodalGradientProcessSettings::SaveAsHistoricalVariable> GradientType;
+            GradientType process = GradientType(this_model_part, DISTANCE, DISTANCE_GRADIENT);
+            process.Execute();
+
+//             // DEBUG
+//             GiDIODebugGradient(this_model_part);
+
+            const double tolerance = 1.0e-8;
+            KRATOS_CHECK_LESS_EQUAL(std::abs(p_node_1->FastGetSolutionStepValue(DISTANCE_GRADIENT_X)) - 1.0, tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs(p_node_2->FastGetSolutionStepValue(DISTANCE_GRADIENT_X)) - 1.0, tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs(p_node_3->FastGetSolutionStepValue(DISTANCE_GRADIENT_X)) - 1.0, tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs(p_node_5->FastGetSolutionStepValue(DISTANCE_GRADIENT_X)) - 1.0, tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs(p_node_9->FastGetSolutionStepValue(DISTANCE_GRADIENT_X)) - 1.0, tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs(p_node_10->FastGetSolutionStepValue(DISTANCE_GRADIENT_X)) - 1.0, tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs(p_node_11->FastGetSolutionStepValue(DISTANCE_GRADIENT_X)) - 1.0, tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs(p_node_12->FastGetSolutionStepValue(DISTANCE_GRADIENT_X)) - 1.0, tolerance);
+        }
+
+        /**
+        * Checks the correct work of the nodal gradient compute
+        * Test quadrilateral
+        */
+        KRATOS_TEST_CASE_IN_SUITE(NodalGradient3, KratosCoreFastSuite)
+        {
+            Model current_model;
+            
+            ModelPart& this_model_part = current_model.CreateModelPart("Main");
+            this_model_part.SetBufferSize(2);
+            
+            this_model_part.AddNodalSolutionStepVariable(DISTANCE);
+            this_model_part.AddNodalSolutionStepVariable(DISTANCE_GRADIENT);
+            
+            Properties::Pointer p_elem_prop = this_model_part.pGetProperties(0);
+            
+            auto& process_info = this_model_part.GetProcessInfo();
+            process_info[STEP] = 1;
+            process_info[NL_ITERATION_NUMBER] = 1;
+            
+            // First we create the nodes 
+            NodeType::Pointer p_node_1 = this_model_part.CreateNewNode(1, 0.0 , 0.0 , 0.0);
+            NodeType::Pointer p_node_2 = this_model_part.CreateNewNode(2, 1.0 , 0.0 , 0.0);
+            NodeType::Pointer p_node_3 = this_model_part.CreateNewNode(3, 1.0 , 1.0 , 0.0);
+            NodeType::Pointer p_node_4 = this_model_part.CreateNewNode(4, 0.0 , 1.0 , 0.0);
+            NodeType::Pointer p_node_5 = this_model_part.CreateNewNode(5, 2.0 , 0.0 , 0.0);
+            NodeType::Pointer p_node_6 = this_model_part.CreateNewNode(6, 2.0 , 1.0 , 0.0);
+            
+            // Initialize nodal area
+            for (auto& node : this_model_part.Nodes())
```

Comment:
variable-utils?

Reply:
It is a test. The test already take long to compile to add more dependencies